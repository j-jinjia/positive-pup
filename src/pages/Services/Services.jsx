import Layout from "../../components/Layout/Layout";
import "./Services.scss";
import Header from "../../components/Header/Header";
import { CoursesList } from "../../Containers/CoursesList/CoursesList";
import courseData from "../../assets/mockData/courseData";
import ServicesGetInTouchSection from "../../components/ServicesGetInTouchSection/ServicesGetInTouchSection";
import { Link } from "react-router-dom";
import SearchBar from "../../components/SearchBar/SearchBar";
import { useState } from "react";

const Services = () => {
  const [searchTerm, setSearchTerm] = useState("");

  const handleInput = (event) => {
    const cleanInput = event.target.value.toLowerCase();
    setSearchTerm(cleanInput);
  };

  const filteredSearchItems = courseData.filter((course) => {
    const filteredCourseHeading = course.courseHeading.toLowerCase();
    return filteredCourseHeading.includes(searchTerm);
  });

  return (
    <Layout>
      <Header
        isLeftAlign={false}
        headingText={"Services"}
        subheadingText={"Take a peek at everything we offer"}
      />
      <ServicesGetInTouchSection
        buttonText={"GET IN TOUCH"}
        link={<Link to="/get-in-touch">GET IN TOUCH</Link>}
      />
      <SearchBar
        searchTerm={searchTerm}
        handleInput={handleInput}
        label="Search Courses"
      />
      <CoursesList courseData={filteredSearchItems} />
    </Layout>
  );
};

export default Services;
