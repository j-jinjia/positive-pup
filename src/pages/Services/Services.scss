@use "../../styles/variables" as *;

.services {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 5rem;
  text-align: center;
  padding: 1rem;

  &__heading {
    color: $colour-primary;
    font-size: 1rem;
    margin: 1rem;
  }

  &__subheading {
    font-size: 1.5rem;
  }
}
