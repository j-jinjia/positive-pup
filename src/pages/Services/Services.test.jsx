import { customRender } from "../../utils/testUtils";
import Services from "./Services";

it("should render the Services page", () => {
  const { container } = customRender(<Services />);

  expect(container).toMatchSnapshot();
});
