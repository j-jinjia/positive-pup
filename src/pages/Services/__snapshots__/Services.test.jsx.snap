// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Services page 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="navbar"
    >
      <div
        class="navbar__container"
      >
        <a
          href="/"
        >
          <img
            alt="The Positive Pup logo"
            class="navbar__container-logo"
            src="footer-logo.svg"
          />
        </a>
        <nav
          class="navbar__nav"
        >
          <a
            class="navbar__nav-item"
            href="/about"
          >
            About
          </a>
          <a
            class="navbar__nav-item"
            href="/services"
          >
            Services
          </a>
          <a
            class="navbar__nav-item"
            href="/testimonials"
          >
            Testimonials
          </a>
          <a
            class="navbar__nav-item"
            href="/gallery"
          >
            Gallery
          </a>
          <span
            class="navbar__nav-item navbar__nav-item--phone"
          >
            <img
              alt=""
              class="navbar__nav-item-button"
              src="phone.svg"
            />
            <a
              class="navbar__nav-item"
              href="/get-in-touch"
            >
              Get in Touch
            </a>
          </span>
          <div
            class="button button--primary button--link"
          >
            <a
              href="/book-now"
            >
              Book Now
            </a>
          </div>
        </nav>
      </div>
    </div>
    <div
      class="header"
    >
      <h1
        class="header__heading"
      >
         
        Services
         
      </h1>
      <h2
        class="header__subheading"
      >
        Take a peek at everything we offer
      </h2>
    </div>
    <div
      class="services-get-in-touch-section"
    >
      <h3
        class="services-get-in-touch-section__header"
      >
        Have a question about any of our courses?
      </h3>
      <p
        class="services-get-in-touch-section__subtext"
      >
        Please take a look at our FAQ’s to see if it’s already been answered. Alternatively, get in touch using the button below.
      </p>
      <div
        class="button button--secondary button--link"
      >
        <a
          href="/get-in-touch"
        >
          GET IN TOUCH
        </a>
      </div>
    </div>
    <div
      class="category-filter category-filter--desktop"
    >
      <button
        class="category-filter__option category-filter__option--selected"
        value="All"
      >
        All
      </button>
      <button
        class="category-filter__option"
        value="Online Courses"
      >
        Online Courses
      </button>
      <button
        class="category-filter__option"
        value="Group Classes"
      >
        Group Classes
      </button>
      <button
        class="category-filter__option"
        value="Bespoke Packages"
      >
        Bespoke Packages
      </button>
    </div>
    <div
      class="search-bar"
    >
      <label
        class="search-bar__label"
        for="Search Courses"
      >
        Search Courses
      </label>
      <input
        class="search-bar__input"
        id="Search Courses"
        role="input"
        type="text"
        value=""
      />
      <img
        alt="search icon"
        class="search-bar__input-icon"
        src="searchBarIcon.svg"
      />
    </div>
    <div
      class="courses-list"
    >
      <div
        class="course-card"
      >
        <div
          class="course-card__img-container"
        >
          <img
            alt="dog"
            class="course-card__img"
            src="image.png"
          />
        </div>
        <div
          class="course-card__course-content"
        >
          <p
            class="course-card__course-type"
          >
            Group Class
          </p>
          <p
            class="course-card__course-heading"
          >
            Little Stars Course
          </p>
          <div
            class="course-card__labels"
          >
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                In Person
              </span>
            </div>
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                10 weeks+
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="course-card"
      >
        <div
          class="course-card__img-container"
        >
          <img
            alt="dog"
            class="course-card__img"
            src="image1.png"
          />
        </div>
        <div
          class="course-card__course-content"
        >
          <p
            class="course-card__course-type"
          >
            Group Class
          </p>
          <p
            class="course-card__course-heading"
          >
            Rising Stars Course
          </p>
          <div
            class="course-card__labels"
          >
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                In Person
              </span>
            </div>
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                16 weeks+
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="course-card"
      >
        <div
          class="course-card__img-container"
        >
          <img
            alt="dog"
            class="course-card__img"
            src="image2.png"
          />
        </div>
        <div
          class="course-card__course-content"
        >
          <p
            class="course-card__course-type"
          >
            Bespoke Package
          </p>
          <p
            class="course-card__course-heading"
          >
            1:1 Personalised Training
          </p>
          <div
            class="course-card__labels"
          >
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                In Person
              </span>
            </div>
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                Intensive
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="course-card"
      >
        <div
          class="course-card__img-container"
        >
          <img
            alt="dog"
            class="course-card__img"
            src="image.png"
          />
        </div>
        <div
          class="course-card__course-content"
        >
          <p
            class="course-card__course-type"
          >
            Group Class
          </p>
          <p
            class="course-card__course-heading"
          >
            Little Stars Course
          </p>
          <div
            class="course-card__labels"
          >
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                Online
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="course-card"
      >
        <div
          class="course-card__img-container"
        >
          <img
            alt="dog"
            class="course-card__img"
            src="image1.png"
          />
        </div>
        <div
          class="course-card__course-content"
        >
          <p
            class="course-card__course-type"
          >
            Online Course
          </p>
          <p
            class="course-card__course-heading"
          >
            Chaos to Calm
          </p>
          <div
            class="course-card__labels"
          >
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                Online
              </span>
            </div>
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                Flexible
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="course-card"
      >
        <div
          class="course-card__img-container"
        >
          <img
            alt="dog"
            class="course-card__img"
            src="image3.png"
          />
        </div>
        <div
          class="course-card__course-content"
        >
          <p
            class="course-card__course-type"
          >
            Online Course
          </p>
          <p
            class="course-card__course-heading"
          >
            Rapid Recall
          </p>
          <div
            class="course-card__labels"
          >
            <div
              class="card-label"
            >
              <span
                class="card-label__text"
              >
                Online
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="footer__image-container"
      >
        <img
          alt="logo"
          class="footer__logo"
          src="footer-logo.svg"
        />
      </div>
      <ul
        class="footer__links-container"
      >
        <li
          class="footer__list-item"
        >
          <a
            class="footer__link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="footer__list-item"
        >
          <a
            class="footer__link"
            href="/services"
          >
            Services
          </a>
        </li>
        <li
          class="footer__list-item"
        >
          <a
            class="footer__link"
            href="/gallery"
          >
            Gallery
          </a>
        </li>
        <li
          class="footer__list-item footer__list-item--desktop"
        >
          <a
            class="footer__link"
            href="/terms-and-conditions"
          >
            Terms & Conditions
          </a>
        </li>
        <li
          class="footer__list-item footer__list-item--mobile"
        >
          <a
            class="footer__link"
            href="/terms-and-conditions"
          >
            T&C's
          </a>
        </li>
        <li
          class="footer__list-item"
        >
          <a
            class="footer__link"
            href="/faqs"
          >
            FAQ's
          </a>
        </li>
        <li
          class="footer__list-item"
        >
          <a
            class="footer__link"
            href="/get-in-touch"
          >
            Contact
          </a>
        </li>
      </ul>
      <hr
        class="footer__line-break"
      />
      <div
        class="footer__secondary-links-container"
      >
        <div
          class="footer__copyright-text"
        >
          © 2022 The Positive Pup. All rights reserved.
        </div>
        <div
          class="footer__social-links"
        >
          <a
            href="https://www.facebook.com/thepositivepupdogtraining"
            target="blank"
          >
            <img
              alt="link to facebook"
              src="facebook-logo.svg"
            />
          </a>
          <a
            href="https://www.instagram.com/the_positivepup/"
            target="blank"
          >
            <img
              alt="link to instagram"
              src="instagram-logo.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
